Docs: add `WHEN NOT MATCHED BY SOURCE` to Spark `MERGE INTO` doc (#11636)
apache/iceberg: docs/docs/spark-writes.md

Only one record in the source data can update any given row of the target table, or else an error will be thrown.

> Starting from Spark 3.5, you can use `WHEN NOT MATCHED BY SOURCE ... THEN ...` to update or delete rows that are not present in the source data:

Reviewer: ```suggestion
Spark 3.5 added support for `WHEN NOT MATCHED BY SOURCE ... THEN ...` to update or delete rows that are not present in the source data:
```
